backend/middleware/exception.py

```diff
@@ -68,13 +68,17 @@ def process_exception(
         # Handle only when running in production, for debug mode
         # Django takes care of displaying a detailed HTML page.
         if not settings.DEBUG and exception:
+            logger.error(f"Unhandled exception by DRF for {request}, logging error...")
+            ExceptionLoggingMiddleware.format_exc_and_log(
+                request=request, exception=exception
+            )
             detail = {"detail": "Error processing the request."}
```

Comment:
@chandrasekharan-zipstack @muhammad-ali-e this response format might need to be changed atleast in case of API deployment responses. Can be taken up separately.